I2B2UI-912/913: [Agg Table] Add date range constraint to rows

hackerceo opened the pull request.

marcdanie reviewed (changes requested):
Hi Nick. I noticed a few of issues. 

1. For the date constraint, if I set a date then the Save button is enabled. However, if I clear the date constraint the Save button is still enabled. I would have expected the Save button to be disabled again.
2. I am able to manually enter invalid dates and save. The date link ends up with NaN's in : 
<img width="1053" alt="Screenshot 2024-10-25 at 3 31 32 PM" src="https://github.com/user-attachments/assets/1710c94c-7f1a-4601-87fa-4ea05a44245f">

3.  If the value constraint link is too long then the Set Date link is cutoff in the UI. I can expand the width of the plugin to see more of the text. Maybe putting the value and date constraint links on two separate lines might help. Just a thought?
<img width="981" alt="Screenshot 2024-10-25 at 3 22 14 PM" src="https://github.com/user-attachments/assets/19760ebb-98ca-45e6-88d3-c3be2b5f2e64">

hackerceo commented:
To fix #1 fully requires additional state management (at the parent level?) to save the original state of the dates.  I have removed the flawed implementation. If need be, users can request that enhancement ticket.

I fixed #2, the save button now becomes disabled when either date picker is failing validation.

Regarding #3, it was stated that we implement this as it was in the existing design, which uses a single line.  Adding a second line would require that all lines have their height expanded regardless of if their constrains take up 2 lines or not.  We should talk to the stakeholders to see if this is what they want (or if they are even affected by this).
